### PR TITLE
Add logging to clone/mount paths

### PR DESCRIPTION
**Pull request by wilbaker**
Part of the work required for #1205

- Add additional logging to help diagnose hangs during clone/mount

### CODE REVIEW

File: GVFS/GVFS.Platform.POSIX/ProcessLauncher.cs [hunk 1]
```diff
@@ -54,40 +56,29 @@ public static void StartBackgroundProcess(ITracer tracer, string programName, st
...
+            // daemons.
+            //
+            // If nochdir is zero, daemon() changes the process's current working
+            // directory to the root directory("/"); otherwise, the current working
+            // directory is left unchanged.
+            //
+            // If noclose is zero, daemon() redirects standard input, standard
+            // output and standard error to /dev/ null; otherwise, no changes are
+            // made to these file descriptors.
+            if (Daemon(nochdir: 1, noclose: 0) != 0)
```
Comment (jeschu1) [inline GVFS/GVFS.Platform.POSIX/ProcessLauncher.cs]: Using Daemon is just a good general cleanup aside from the extra logging correct?  Was there any other reason for this?

Comment (wilbaker) [reply GVFS/GVFS.Platform.POSIX/ProcessLauncher.cs]: It was just to move us one step closer to standard methods/functions and away from our own custom solution.

However, this is not strictly required for the logging and so I might just take that change out for now and focus on getting us onto `Process.Start`

Comment (wilbaker) [reply GVFS/GVFS.Platform.POSIX/ProcessLauncher.cs]: @jeschu1 I decided to back this change out for now and just go with the extra logging.  I've leave the launching code as-is until we have a better fix for the hang.

**Pull request merged**